src/functions.c

```diff
@@ -1009,7 +1018,7 @@ sds functionsCreateWithLibraryCtx(sds code, int replace, sds* err, functionsLibC
 error:
     if (iter) dictReleaseIterator(iter);
     if (new_li) engineLibraryFree(new_li);
-    if (old_li) libraryLink(lib_ctx, old_li);
+    if (old_li && !old_li->is_linked) libraryLink(lib_ctx, old_li);
```

Comment:
can't we just implicitly fail `libraryLink` if the library is already there?
i.e. we already have ways to know if it's linked or not, don't we?
e.g.  `dictAdd(lib_ctx->libraries, li->name, li);` would fail.

Reply:
I thought it will be safer to make it explicit to avoid bugs, i.e, if we unlink/link a library it is supposed to be link/unlinked. If you prefer the other approach I will change it.

Reply:
it looks odd to me that redundancy. i.e. two ways to check for the same thing, and the potential they'll be conflicting.

the only problem was that we remove and then restore a library, right?
when we removed it, why didn't we also unregister it?
or was the problem that we got to the `error` label before removing? (in which case maybe a variable on the stack is enough)

Reply:
We got the error before removing it, OK I will add a stack variable.

Reply:
@oranagra update the PR, easier one line fix :)